tests/integration_tests/vectorstores/test_mongodb_atlas.py

```diff
@@ -0,0 +1,100 @@
+"""Test MongoDB Atlas Vector Search functionality."""
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING
+
+import pytest
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.vectorstores.mongodb_atlas import MongoDBAtlasVectorSearch
+
+if TYPE_CHECKING:
+    from pymongo import MongoClient
+
+INDEX_NAME = "langchain-test-index"  # name of the index
+NAMESPACE = "langchain_test_db.langchain_test_collection"  # name of the namespace
+CONNECTION_STRING = os.environ.get("MONGODB_ATLAS_URI")
+
+
+@pytest.mark.requires("pymongo")
+class TestMongoDBAtlasVectorSearch:
+    @classmethod
+    def test_collection(cls) -> MongoClient:
+        from pymongo import MongoClient
+
+        db_name, collection_name = NAMESPACE.split(".")
+        client = MongoClient(CONNECTION_STRING)
```

Comment:
Each time a MongoClient is instantiated, a whole connection pool is created (+ authentication, etc.).